docs/samples/validation/attributes/Program.cs

```diff
@@ -3,8 +3,10 @@
 
 using System;
 using System.ComponentModel.DataAnnotations;
+using System.Linq;
 using McMaster.Extensions.CommandLineUtils;
 
+[RequiredDependentsValidation(nameof(Attachments), nameof(MaxSize))]
```

Comment:
This sample was a little hard to follow.  I think what you're trying to do is assert that if `--max-size` is specified, `--attachments` must also have been given, right?

If so, rather than writing an abstract validation attribute, let's make this validation attribute very specific to what we want to validate. It makes it easier to understand. I'm thinking something like `[MaxSizeOptionRequiresAttachments]` with an implementation like

```
protected override ValidationResult IsValid(object value, ValidationContext context)
{
   if (value is AttributeProgram obj)
   {
      	if (obj.MaxSize.HasValue && Attachments == null || Attachments.Length == 0)
     	{
        	return new ValidationResult("--max-size cannot be used unless --attachments is also specified");
    	}
   	}
	return ValidationResult.Success;
}
```

Reply:
Regarding the sample, I can see you point. If it was for real life I would make something more reusable. I'll make this change.